lib/modules/csv_exporter.rb

```diff
@@ -1,32 +1,180 @@
+        result.find_answer_by_key_value("short_name", "seek_no_harm", default),
+        result.find_answer_by_key_value("short_name", "giving_voice_ethical", default),
+        result.find_answer_by_key_value("short_name", "economic_growth", default),
+        result.find_answer_by_key_value("short_name", "communities_manage", default),
+        result.find_answer_by_key_value("short_name", "win_win", default),
+        result.find_answer_by_key_value("short_name", "rationales_weakens", default),
+        result.find_answer_by_key_value("short_name", "emphasise_value", default),
+        result.find_answer_by_key_value("short_name", "doom_gloom", default),
+        result.find_answer_by_key_value("short_name", "biological_diversity", default),
+        result.find_answer_by_key_value("short_name", "severe_perturbations", default),
+        result.find_demographic_answer_by_key("age", default),
+        result.find_demographic_answer_by_key("gender", default),
+        result.find_demographic_answer_by_key("education", default),
+        result.find_demographic_answer_by_key("educational_specialism", default),
+        result.find_demographic_answer_by_key("country", default),
+        self.format_multiple_answer(result.find_demographic_answer_by_key("location_of_work"), default),
+        self.format_multiple_answer(result.find_demographic_answer_by_key("conservation_sectors"), default),
+        result.find_demographic_answer_by_key("experience_in_other_fields", default),
+        self.format_multiple_answer(result.find_demographic_answer_by_key("non_conservation_sectors"), default),
+        result.find_demographic_answer_by_key("professional_engagement", default),
+        result.find_demographic_answer_by_key("seniority_of_current_role", default),
+        result.find_demographic_answer_by_key("category_of_most_professional_work", default),
+        result.find_demographic_answer_by_key("researcher_experience", default),
+        self.format_multiple_answer(result.find_demographic_answer_by_key("value_shaping_items"), default),
+        result.find_demographic_answer_by_key("shaping_values", default),
+        result.find_demographic_answer_by_key("email", default)
+      ].flatten
+
+      row.each do |answer|
+        # For text answers, replace commas
+        answer.gsub(",", "") if answer.is_a?(String)
+      end
+    end
+
+    def self.json_headers_for(result, question_type)
+      headers = []
+      result.answers[question_type].each {|q| headers << q["question_text"] || ""}
+      headers
+    end
+
+    def self.json_row_for(result, question_type)
+      row = []
+      result.answers[question_type].each {|q| row << q["answer_inputted"] || ""}
+      row
+    end
+
+    def self.format_multiple_answer(answer, default)
+      return default if answer.nil?
+      answer = Array.wrap(answer.values) if answer.is_a?(Hash)
+      answer = answer.reject(&:empty?).join("|")
```

Comment:
Is there any chance one of the answers could be `nil`? If yes, I think we'd be better off with `answer.reject(&:blank?).join("|")` here, as `empty?` is not a method for `nil`, and it'd blow everything up.